Sources/StreamChat/WebSocketClient/WebSocketClient.swift

```diff
@@ -185,8 +185,14 @@ class WebSocketClient {
     @objc private func handleAppDidEnterBackground() {
         guard options.contains(.staysConnectedInBackground), connectionState.isActive else { return }
         
-        let backgroundTask = backgroundTaskScheduler?.beginBackgroundTask { [weak self] in
+        var backgroundTask: UIBackgroundTaskIdentifier?
+        backgroundTask = backgroundTaskScheduler?.beginBackgroundTask { [weak self] in
             self?.disconnect(source: .systemInitiated)
+            // We need to call `endBackgroundTask` else our app will be killed
+            if let backgroundTask = backgroundTask {
+                self?.backgroundTaskScheduler?.endBackgroundTask(backgroundTask)
+            }
+            backgroundTask = .invalid
```

Comment:
Don't we have tests that cover this logic? Would be nice add tests, so there are no more regressions

Reply:
How can we simulate this? 😓

Reply:
We already have tests for the disconnect 🤔 I guess we just need to verify if `backgroundTaskScheduler?.endBackgroundTask` is called if there's a background task?